src/state/finishedGames/finishedGameReducer.ts

```diff
@@ -11,29 +11,30 @@ const finishedGameReducer = ( state = initialState, action: GenericAction ) => {
     switch (action.type) {
 
         case SAVE_GAME_DATA: {
-            let clone = cloneState( state );
-
+            const clone = cloneState( state );
             clone.push( action.payload );
             return clone;
         }
         case SAVE_GAME_DATA_PENDING: {
-            let clone = cloneState( state );
+            const clone = cloneState( state );
             clone[clone.length - 1].saveState = 'pending';
             return clone;
         }
         case SAVE_GAME_DATA_REJECTED: {
             let clone = cloneState( state );
             // we don't know for sure actually if the last element is the element this corresponds to.
-            // id should be introduced later
-            let lastElement = clone[clone.length - 1];
+            // client id should be introduced later
+            const lastElement = clone[clone.length - 1];
             lastElement.saveState = 'rejected';
             lastElement.errorMessage = action.payload;
             clone[clone.length - 1] = lastElement;
             return clone;
         }
         case SAVE_GAME_DATA_FULFILLED: {
-            let clone = cloneState( state );
-            clone[clone.length - 1].saveState = 'fulfilled';
+            const clone = cloneState( state );
+            const lastElement = clone[clone.length - 1];
+            lastElement.saveState = 'fulfilled';
+            lastElement.id = action.payload;
```

Comment:
@RohrerF Id is set here when request has been fulfilled. We don't know the id before it was inserted.